### ISSUE TITLE
Can't seem to get the update mods to work

### ISSUE THREAD

**Issue by Flukethoughts** [FezVrasta/ark-server-tools#627]
commands used
sudo arkmanager update --update-mods
sudo arkmanager update --update-mods --warn

But when I join the server I get mod mismatch

**Comment by klightspeed** [comment]
What is the output of `arkmanager --version`?

What is the output of `arkmanager --update-mods`?

**Comment by Flukethoughts** [comment]
Version: 1.6
Channel: master
Commit: 2bc099f
Release Tag: v1.6.04
Blob SHA: bcac987ecdf8eb48d805367185c729a79ff3b250

flukethoughts@tartarus:~$ sudo arkmanager --update-mods
Running command '--update-mods' for instance 'main'
arkmanager v1.6: unknown command '--update-mods' specifiedUsage: arkmanager [Commands]

Commands can be followed by one or more @instance arguments
The special '@all' instance selects all instances
Commands may also be followed by zero or more --options

**Comment by Flukethoughts** [comment]
flukethoughts@tartarus:~$ sudo arkmanager update --update-mods @all
Running command 'update' for instance 'shingo'
2016-10-02 11:50:24: checking for update; PID: 1504
Your server is already up to date! The most recent version is 1366578.
Running command 'update' for instance 'Mortem'
2016-10-02 11:50:40: checking for update; PID: 1998
Your server is already up to date! The most recent version is 1366578.
Running command 'update' for instance 'island'
2016-10-02 11:50:56: checking for update; PID: 2489
Your server is already up to date! The most recent version is 1366578.
Running command 'update' for instance 'Madagascar'
2016-10-02 11:51:12: checking for update; PID: 3006
Your server is already up to date! The most recent version is 1366578.
Running command 'update' for instance 'main'
2016-10-02 11:51:28: checking for update; PID: 3522
Your server is already up to date! The most recent version is 1366578.
Running command 'update' for instance 'scorched'
2016-10-02 11:51:43: checking for update; PID: 3988
Your server is already up to date! The most recent version is 1366578.

**Comment by Flukethoughts** [comment]
main.cfg

# config environment
arkserverroot="/home/steam/Center"                                     # path of your ARK server files (default ~/ARK)

# ARK server options - use ark_<optionname>=<value>
# comment out these values if you want to define them
# inside your GameUserSettings.ini file
serverMap="TheCenter"                                               # server map (default TheIsland)
#serverMapModId="661564024"                                         # Uncomment this to specify the Map Mod Id (<fileid> in http://steamcommunity.com/sharedfiles/filedetails/?id=<fileid>)
#ark_TotalConversionMod="496735411"                                 # Uncomment this to specify a total-conversion mod
ark_RCONEnabled="false"                                              # Enable RCON Protocol
ark_RCONPort="32330"                                                # RCON Port
#ark_SessionName="Moddedark.com The Islands"                                  # if your session name needs special characters please use the .ini instead
ark_Port="7778"                                                     # ARK server port (default 7778)
ark_QueryPort="27015"                                               # ARK query port (default 27015)
ark_ServerPassword=""                                               # ARK server password, empty: no password required to login
ark_ServerAdminPassword=""                                          # ARK server admin password, KEEP IT SAFE!
ark_MaxPlayers="70"
ark_GameModIds="538986229,516096294,554678442,558651608,561820195,566887000,569786012,575017270,622595587,622608522,630601751,633831197,655261420,681259857,658667129,689838254,693416678,698238991,707081776,729352919,725398419,731604991,747539326,748358288,749145707,751991809,75488$
arkflag_UseBattlEye=true
arkflag_insecure=true
arkflag_NoTransferFromFiltering=true
arkopt_clusterid=Cluster1
#ark_AltSaveDirectoryName="SotF"                                    # Uncomment to specify a different save directory name

# ARK server flags - use arkflag_<optionname>=true
#arkflag_OnlyAdminRejoinAsSpectator=true                            # Uncomment to only allow admins to rejoin as spectator
#arkflag_DisableDeathSpectator=true                                 # Uncomment to disable players from becoming spectators when they die

# ARK server options - i.e. for -optname=val, use arkopt_optname=val
#arkopt_StructureDestructionTag=DestroySwampSnowStructures

**Comment by Flukethoughts** [comment]
arkmanager.cfg
  GNU nano 2.7.0                                                                                                                      File: arkmanager.cfg

arkstChannel="master"                                               # change it to a different branch to get non-stable versions
install_bindir="/usr/local/bin"
install_libexecdir="/usr/local/libexec/arkmanager"
install_datadir="/usr/local/share/arkmanager"

# config SteamCMD
steamcmdroot="/home/steam/steamcmd"                                 # path of your steamcmd instance
steamcmdexec="steamcmd.sh"                                          # name of steamcmd executable
steamcmd_user="steam"                                               # name of the system user who own steamcmd folder
steamcmd_appinfocache="/home/steam/Steam/appcache/appinfo.vdf"      # cache of the appinfo command
#steamlogin="anonymous"                                             # Uncomment this to specify steam login instead of using anonymous login

# config environment
arkserverexec="ShooterGame/Binaries/Linux/ShooterGameServer"        # name of ARK server executable
arkbackupdir="/home/steam/ARK-Backups"                              # path to backup directory
arkwarnminutes="60"                                                 # number of minutes to warn players when using update --warn
arkautorestartfile="ShooterGame/Saved/.autorestart"                 # path to autorestart file
arkAutoUpdateOnStart="false"                                        # set this to true if you want to always update before startup
arkBackupPreUpdate="false"                                          # set this to true if you want to perform a backup before updating
arkStagingDir="/home/steam/ARK-Staging"                            # Uncomment to enable updates to be fully downloaded before restarting the server (reduces downtime while updating)

# Options to automatically remove old backups to keep backup size in check
# Each compressed backup is generally about 1-2MB in size.
arkMaxBackupSizeMB="500"                                            # Set to automatically remove old backups when size exceeds this limit
#arkMaxBackupSizeGB="2"                                             # Uncomment this and comment the above to specify the limit in whole GB

# Update warning messages
# Modify as desired, putting the %d replacement operator where the number belongs
msgWarnUpdateMinutes="This ARK server will shutdown for an update in %d minutes"
msgWarnUpdateSeconds="This ARK server will shutdown for an update in %d seconds"
msgWarnRestartMinutes="This ARK server will shutdown for a restart in %d minutes"
msgWarnRestartSeconds="This ARK server will shutdown for a restart in %d seconds"
msgWarnShutdownMinutes="This ARK server will shutdown in %d minutes"
msgWarnShutdownSeconds="This ARK server will shutdown in %d seconds"
msgWarnCancelled="Restart cancelled by player request"

# Restart cancel chat command
#chatCommandRestartCancel="/cancelupdate"

# ARK server common options - use ark_<optionname>=<value>
# comment out these values if you want to define them
# inside your GameUserSettings.ini file
serverMap="TheIsland"                                               # server map (default TheIsland)
#serverMapModId="469987622"                                         # Uncomment this to specify the Map Mod Id (<fileid> in http://steamcommunity.com/sharedfiles/filedetails/?id=<fileid>)
#ark_TotalConversionMod="496735411"                                 # Uncomment this to specify a total-conversion mod
ark_RCONEnabled="False"                                             # Enable RCON Protocol
ark_ServerPassword=""                                               # ARK server password, empty: no password required to login
ark_ServerAdminPassword=""                               # ARK server admin password, KEEP IT SAFE!
ark_MaxPlayers="70"
#ark_GameModIds="487516323,487516324,487516325"                     # Uncomment to specify additional mods by Mod Id separated by commas
#ark_AltSaveDirectoryName="SotF"                                    # Uncomment to specify a different save directory name

# ARK server flags - use arkflag_<optionname>=true
#arkflag_OnlyAdminRejoinAsSpectator=true                            # Uncomment to only allow admins to rejoin as spectator
#arkflag_DisableDeathSpectator=true                                 # Uncomment to disable players from becoming spectators when they die

# ARK server options - i.e. for -optname=val, use arkopt_optname=val
#arkopt_StructureDestructionTag=DestroySwampSnowStructures

# config Service
logdir="/var/log/arktools"                                          # Logs path (default /var/log/arktools)

# steamdb specific
appid=376030                                                        # Linux server App ID
mod_appid=346110                                                    # App ID for mods

# Mod OS Selection
mod_branch=Windows
# Add mod-specific OS selection below:
#mod_branch_496735411=Windows

# alternate configs
# example for config name "ark1":
#configfile_ark1="/path/to/config/file"

defaultinstance="main"

**Comment by klightspeed** [comment]
Please try upgrading arkmanager to v1.6.05, and see if that fixes your issue.

**Comment by Flukethoughts** [comment]
First
flukethoughts@tartarus:/etc/arkmanager$ sudo arkmanager upgrade-tools
arkmanager v1.6: Checking for updates...
Getting latest release...
A new version has been released: v1.6.05
* Remove a temp file and prevent it from being picked back up
* Add `arkCpuAffinity` setting
* Fix an error in `isModUpdateAvailable`
Do you want to upgrade to v1.6.05? [Y/N] y
Getting commit for latest release...
Downloading installer
Running install.sh
Ark server will now start on boot, if you want to remove this feature run the following line
systemctl disable arkmanager.service
A previous version of ARK Server Tools was detected in your system, your old configuration was not overwritten. You may need to manually update it.
A copy of the new configuration file was included in /etc/arkmanager. Make sure to review any changes and update your config accordingly!
ARK Server Tools successfully upgraded
Version: 1.6
Channel: master
Commit: 4a2699b
Release Tag: v1.6.05
Blob SHA: 2a3301242ca4469f9d320f51210d31ead01c7adc

Okay now updated
flukethoughts@tartarus:/etc/arkmanager$ sudo arkmanager update --update-mods @all
Running command 'update' for instance 'shingo'
2016-10-02 12:09:42: checking for update; PID: 15555
Downloading mod 516096294 ... Mod 516096294 downloaded
Downloading mod 538986229 ... Mod 538986229 downloaded
Downloading mod 554678442 ... Mod 554678442 downloaded
Downloading mod 558651608 ... Mod 558651608 downloaded
Downloading mod 561820195 ... Mod 561820195 downloaded
Downloading mod 566887000 ... Mod 566887000 downloaded
Downloading mod 569786012 ... Mod 569786012 downloaded
Downloading mod 575017270 ... Mod 575017270 downloaded
Downloading mod 622595587 ... Mod 622595587 downloaded
Downloading mod 622608522 ... Mod 622608522 downloaded
Downloading mod 630601751 ... Mod 630601751 downloaded
Downloading mod 633831197 ... Mod 633831197 downloaded
Downloading mod 655261420 ... Mod 655261420 downloaded
Downloading mod 658667129 ... Mod 658667129 downloaded
Downloading mod 661564024 ... Mod 661564024 downloaded
Downloading mod 681259857 ... Mod 681259857 downloaded
Downloading mod 689838254 ... Mod 689838254 downloaded
Downloading mod 693416678 ... Mod 693416678 downloaded
Downloading mod 698238991 ... Mod 698238991 downloaded
Downloading mod 707081776 ... Mod 707081776 downloaded
Downloading mod 725398419 ... Mod 725398419 downloaded
Downloading mod 729352919 ... Mod 729352919 downloaded
Downloading mod 731604991 ... Mod 731604991 downloaded
Downloading mod 747539326 ... Mod 747539326 downloaded
Downloading mod 748358288 ... Mod 748358288 downloaded
Downloading mod 749145707 ... Mod 749145707 downloaded
Downloading mod 751991809 ... Mod 751991809 downloaded
Downloading mod 754885087 ... Mod 754885087 downloaded
Downloading mod 758608778 ... Mod 758608778 downloaded
Copying to staging directory
Downloading ARK update ... The server is already stopped
Applying update from staging directory
Updating mod 698238991
Copying files to /home/steam/Shingo/ShooterGame/Content/Mods/698238991
Updating mod 751991809
Copying files to /home/steam/Shingo/ShooterGame/Content/Mods/751991809

And it's doing this with the other servers as well will let you kno when i start it if it worked or not

**Comment by Flukethoughts** [comment]
also stats if your interested
https://tartarus.flukethoughts.com/netdata/

**Comment by Flukethoughts** [comment]
okay to confirm that fixed worked wonderfully thank you

**Comment by Flukethoughts** [comment]
@klightspeed thank you

**Comment by Flukethoughts** [comment]
okay new issue with the update of mods if I run a cronjob every 15 mins with the amount of servers I have i run into issues.  Is their anyways to have it check the mods of one server and if their is an update push all the updates to the rest instead of checking each one individually because it overlaps and this is an issue when all the servers have the exact same mods it will continue to go through each server and redownload it over and over again. I know this is by design but it is eating up cpu in my case when the process begins to overlap and it doesn't finish the first update before starting the next. 

Cron task every 15 mins checks updates and updates mods
15 mins later it still hasn't completed the first task with 6 servers
Cron task every 15 mins checks updates and updates mods
15 mins later now it's overlapping with the third about to start
so on and so on the server cpu starts to rise heavlly

Things I have done to combat this cpu usage 
I had a setting for forcerespawn dinos and I have removed it on each server their is no real reason to keep it.

However, the task at hand is all th servers use the same mods so the staging area should be able to download the updates then push the updates to all at the same time to prevent the time delay of check each one individually. Other words it might be best to setup a staging mod folder and have all the mods stored in it and then pushed to the instances on updates the update-mods would have its own folder downloaded and checked then pushed to the remaining instances.

@klightspeed  I hope this makes sense sorry If i got confusing their.

**Comment by klightspeed** [comment]
Commit fc4d644 now allows install-cronjob to create a single cronjob that executes a command against multiple instances.

If all of your cronjobs are the same, you should be able to use `crontab -e` to modify the first cronjob, replacing the instance name with `@all`, and remove the other cronjobs that execute the same command against the other instances.
